### ISSUE TITLE
Too Many Silverfish Blocks!

### ISSUE THREAD

**Issue by Wasthereonce** [Greymerk/minecraft-roguelike#56]
**The issue:**
I was playing on an LAN server with my friends, using your mod in a private custom modpack. One of those mods was Minefactory Reloaded, which is part of the problem we ran into. When a silverfish spawned on the chaotic mossy floor from the spawners in the walls, it fell into the MFR poison liquid. As the silverfish swam around in there, it awoken hell from all the silverfish already inside the blocks. The FPS dropped to 1 as we frantically scurried out of the crumbling room. While in the F3 screen, I saw there were over 2,500 entities. No surprise, they were silverfish. I suggest you remove a large majority of all those silverfish blocks. I know what you were thinking. If a trapped chest was triggered there, it would explode and cause a mass of silverfish to swarm you. This doesn't work because the TNT usually kills the silverfish. And what silverfish it doesn't kill reenter a block.

**TL;DR:** Remove 95% of silverfish blocks. Constant damage to silverfish in numerous forms (Glenn's Gases, Minefactory Reloaded, Cactus, etc.) can slow the game to a crash.

**Comment by Greymerk** [comment]
I get the feeling you're describing an older version. The dungeons currently have about 1200 silverfish in total. what you're describing is an event in a single room with 2500 (double the number which you'll normally see in a single dungeon).

In testing I actually found there were far fewer silverfish blocks than i'd even intended, and there's no way i'm about to lower the current number of them.

What version of the mod are you even using?

**Comment by Wasthereonce** [comment]
It was the latest 1.7.10 1.5.0b version. It happened in an intersection too. Now that you mention it, I'm also using a mod called Ex Compressum, which adds Compressed variants of mobs, including silverfish. When they die, they split into 9 regular silverfish. But the silverfish weren't dying because of the poison from MFR, they were just spawning more and more from everywhere. Maybe all the silverfish leaving and entering the blocks caused even more lag. Anyway, when I reentered the world after the crash, the blocks weren't loaded. So all I saw were the entities. I stared at this huge mass of silverfish. As for the number of silverfish, I can't remember if I saw the overall entity count as 2500 in F3 or just the silverfish below I was looking at. But I remember the 2500 specifically. I think some silverfish blocks can spawn 2 silverfish as well, so that could be taken into consideration.

**Comment by Greymerk** [comment]
According to the minecraft wiki the monster egg block will spawn exactly one silverfish when broken. This agrees with my own experience of the way monster egg blocks work.

**Comment by Wasthereonce** [comment]
It might be another mod that adds this behavior
